### DISCUSSION TITLE
retrieving the latest time-based partition that was actually materialized for an asset

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#15194] [Q&A] [answered]
It's possible to skip partitions using the `should_execute` param on a schedule - and so for scheduled partitions, it can be useful to know which partitions were actually materialized. If you want to retrieve the latest partition time-wise that was actually materialized, you can't use `DagsterInstance.get_latest_materialization_event`, because a backfill could potentially skew those results. 

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1689005786157169?thread_ts=1689005786.157169&cid=C01U954MEER).*

**Comment by sryza** [comment, marked as answer]
Here's some code for doing this:

```python
def get_last_materialized_partition(instance, asset_def):
    asset_key = asset_def.key
    materialized_partitions = instance.get_materialization_count_by_partition([asset_key])[
        asset_key
    ].keys()
    ordered_materialized_partitions = [
        partition
        for partition in asset_def.partitions_def.get_partition_keys()
        if partition in materialized_partitions
    ]
    return ordered_materialized_partitions[-1]


# verify it works
if __name__ == "__main__":
    from dagster import DagsterInstance, materialize, asset, DailyPartitionsDefinition

    @asset(partitions_def=DailyPartitionsDefinition(start_date="2023-01-01"))
    def asset1():
        ...

    instance = DagsterInstance.ephemeral()
    materialize([asset1], partition_key="2023-03-03", instance=instance)
    materialize([asset1], partition_key="2023-03-02", instance=instance)
    assert get_last_materialized_partition(instance, asset1) == "2023-03-03"
```

**Reply by Tazoeur** [reply]
`get_materialization_count_by_partition` does not exist anymore (>= 1.5.9, maybe before) you'll have to use `get_materialized_partitions`
